### PR TITLE
01 alexander yordanov

### PR DESCRIPTION
**Pull request by Alexander1022**

### CODE REVIEW

File: homework_03/01_alexander_yordanov/main_task.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+class Point: 
+    def __init__(self, x, y) -> None:
+        assert isinstance(x, float)
+        assert isinstance(y, float)
+
+        self.x = x
+        self.y = y
+
+class Circle:
+    def __init__(self, center, r) -> None:
```
Comment (Martincho2003) [inline homework_03/01_alexander_yordanov/main_task.py:19]: You could use radius instead of r but not a big deal.

File: homework_03/01_alexander_yordanov/main_task.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        self.center = center
+        self.r = r
+
+    def find_relative_position(self, another_circle):
+        assert isinstance(another_circle, Circle)
+
+        distance = math.sqrt((another_circle.center.x - self.center.x) ** 2 + (another_circle.center.y - self.center.y) ** 2)
+        radiusSum = self.r + another_circle.r
+        radiusSub = self.r - another_circle.r
```
Comment (Martincho2003) [inline homework_03/01_alexander_yordanov/main_task.py:30]: Great idea to put them into variables

File: homework_03/01_alexander_yordanov/main_task.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        if distance == 0 and radiusSub == 0:
```
Comment (Martincho2003) [inline homework_03/01_alexander_yordanov/main_task.py:32]: You are not catching all possible variants like this in the picture and maybe more
image

File: homework_03/01_alexander_yordanov/main_task.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            return ReturnTypes.SAME
+
+        elif abs(distance) > radiusSum:
+            return ReturnTypes.NO_COMMON_POINTS
```
Comment (Martincho2003) [inline homework_03/01_alexander_yordanov/main_task.py:36]: That's ok

File: homework_03/01_alexander_yordanov/main_task.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+        elif abs(distance) == radiusSum or abs(distance) == abs(radiusSub):
```
Comment (Martincho2003) [inline homework_03/01_alexander_yordanov/main_task.py:38]: I can't get the logic of `abs(distance) == abs(radiusSub)` so it may be true but also not so much. Maybe the first is enough, but there is a possibility that I just haven't found some cases.

File: homework_03/01_alexander_yordanov/main_task.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            return ReturnTypes.TOUCHING
+
+        elif abs(radiusSub) < abs(distance) < radiusSum:
```
Comment (Martincho2003) [inline homework_03/01_alexander_yordanov/main_task.py:41]: Looks right

File: homework_03/01_alexander_yordanov/tests.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    assert circle3.find_relative_position(circle4) == main_task.ReturnTypes.TOUCHING
+
+# test if cirlces are intersecting each other
+def test_intersecting():
+    circle1 = main_task.Circle(main_task.Point(1.0, 1.0), 1.0)
+    circle2 = main_task.Circle(main_task.Point(1.5, 1.0), 1.0)
+
+    assert circle1.find_relative_position(circle2) == main_task.ReturnTypes.INTERSECTING
+
+test_same_circles()
```
Comment (Martincho2003) [inline homework_03/01_alexander_yordanov/tests.py:50]: You could name the file "test_main_task" and use pytest instead of calling the functions.

**Review by Martincho2003 — DISMISSED** [review]
I want you to look for the examples that you don't handle in the SAME circles and also add maybe add some more tests for each case(may be some edge-cases).

File: homework_03/01_alexander_yordanov/main_task.py (hunk 1 above)
Comment (marto55) [reply homework_03/01_alexander_yordanov/main_task.py:19]: Actually i preffer "r" instead of "radius"

File: homework_03/01_alexander_yordanov/main_task.py (hunk 2 above)
Comment (marto55) [reply homework_03/01_alexander_yordanov/main_task.py:30]: I wouln't glorify it, but it's good idea

File: homework_03/01_alexander_yordanov/main_task.py (hunk 5 above)
Comment (marto55) [reply homework_03/01_alexander_yordanov/main_task.py:38]: the way you implemented it, distance is always positive, you don't have to use abs()

File: homework_03/01_alexander_yordanov/tests.py (hunk 7 above)
Comment (marto55) [reply homework_03/01_alexander_yordanov/tests.py:50]: i think pytest will execute it even if you don't call the functions, because the filename is "test*"

File: homework_03/01_alexander_yordanov/main_task.py (hunk 3 above)
Comment (marto55) [reply homework_03/01_alexander_yordanov/main_task.py:32]: Indeed your implementation isn't write but you already know that

File: homework_03/01_alexander_yordanov/tests.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+def create_circle():
+    center = main_task.Point(3 ,6)
+    r = 2
+    circle = main_task.Circle(center, r)
+    assert isinstance(circle, main_task.Circle)
+
+# test if one circle is on the same place like the other (in this case i use the same circle)
+def test_same_circles():
+    circle1 = main_task.Circle(main_task.Point(1.0, 2.0), 1.0)
+
```
Comment (marto55) [inline homework_03/01_alexander_yordanov/tests.py:19]: tecknically  you need more tests to catch all cases but it doesn't matter you are good

File: homework_03/01_alexander_yordanov/tests.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+def test_intersecting():
+    circle1 = main_task.Circle(main_task.Point(1.0, 1.0), 1.0)
+    circle2 = main_task.Circle(main_task.Point(1.5, 1.0), 1.0)
+
+    assert circle1.find_relative_position(circle2) == main_task.ReturnTypes.INTERSECTING
+
+test_same_circles()
+test_no_common_points()
+test_touching()
+test_intersecting()
```
Comment (marto55) [inline homework_03/01_alexander_yordanov/tests.py:53]: Good job

**Review by marto55 — APPROVED** [review]
You good
Go have sove rest
